splash/tests/test_execute.py

```diff
@@ -3925,3 +3925,433 @@ def test_key_error(self):
+            assert(splash:go(splash.args.url))
+            assert(splash:wait(0.1))
+
+            local p = splash:select('p')
+            local form = splash:select('form#login')
+            local username = splash:select('input[name="username"]')
+            local password = splash:select('input[name="password"]')
+            local title = splash:select('.title')
+
+            return {
+                p=select(2, assert(p:node_property('nodeName'))):lower(),
+                form=select(2, assert(form:node_property('nodeName'))):lower(),
+                username=select(2, assert(username:node_property('nodeName'))):lower(),
+                password=select(2, assert(password:node_property('nodeName'))):lower(),
+                title=select(2, assert(title:node_property('nodeName'))):lower(),
+            }
+        end
+        """, {"url": self.mockurl("various-elements")})
+
+        self.assertStatusCode(resp, 200)
+        self.assertEqual(resp.json(), {
+            "p": "p",
+            "form": "form",
+            "username": "input",
+            "password": "input",
+            "title": "h1"
+        })
+
+    def test_element_ids(self):
+        resp = self.request_lua("""
+        local treat = require('treat')
+        function main(splash)
+            assert(splash:go(splash.args.url))
+            assert(splash:wait(0.1))
+
+            local p = splash:select('p')
+            local form = splash:select('form#login')
+            local username = splash:select('input[name="username"]')
+            local password = splash:select('input[name="password"]')
+            local title = splash:select('.title')
+
+            return treat.as_array({
+                p.id,
+                form.id,
+                username.id,
+                password.id,
+                title.id,
+            })
+        end
+        """, {"url": self.mockurl("various-elements")})
+
+        self.assertStatusCode(resp, 200)
+        ids = resp.json()
+        self.assertEqual(len(ids) == (len(set(ids))), True)
+
+    def test_bad_selector(self):
+        resp = self.request_lua("""
+         function main(splash)
+            assert(splash:go(splash.args.url))
+            assert(splash:wait(0.1))
+
+            local element = splash:select('!notaselector')
+
+            return element:exists()
+        end
+        """, {"url": self.mockurl("various-elements")})
+
+        self.assertStatusCode(resp, 400)
+        err = self.assertScriptError(resp, ScriptError.SPLASH_LUA_ERROR)
+        self.assertEqual(err["info"]["splash_method"], "select")
+
+    def test_node_property_returns_element(self):
+        resp = self.request_lua("""
+        function main(splash)
+            assert(splash:go(splash.args.url))
+            assert(splash:wait(0.1))
+
+            local p = splash:select('p')
+            local ok, parent = p:node_property('parentNode')
+            local ok, remove = parent:node_method('remove')
```

Comment:
There's a branch that would allow `splash:select('p').parentNode:remove()`, that raises errors if something goes wrong, and I have high hopes for it. +1 from me.

Also, while writing it made me wonder if we want to have `.parentNode` for attributes and `:remove` for methods.